bin/build/src/release/version_info.clj

```diff
@@ -40,48 +45,59 @@
   "The info map for the version we're currently releasing to add to the version info file."
   []
   {:version    (str "v" (c/version))
-   :released   (str (java.time.LocalDate/now))
+   :released   (str (LocalDate/now))
    :patch      (c/patch-version?)
    ;; TODO -- these need to be curated a bit before publishing...
    :highlights (mapv :title (github/milestone-issues))})
 
-(defn- generate-version-info! []
-  (let [filename (version-info-filename)
-        tmpname  (tmp-version-info-filename)]
-    (u/step (format "Generate %s" filename)
-      (u/step (format "Delete and create %s" tmpname)
-        (u/delete-file-if-exists! tmpname)
-        (let [{:keys [latest], :as info} (current-version-info)]
-          (spit tmpname (-> info
-                            ;; move the current `:latest` to the beginning of `:older`
-                            (update :older (fn [older]
-                                             (distinct (cons latest older))))
-                            (assoc :latest (info-for-new-version))
-                            json/generate-string)))))))
+(defn- generate-updated-version-info
+  [edition]
+  (u/step (format "Generate %s" (version-info-filename edition))
+    (let [{:keys [latest], :as info} (current-version-info edition)]
+      (-> info
+          ;; move the current `:latest` to the beginning of `:older`
+          (update :older (fn [older]
+                           (distinct (cons latest older))))
+          (assoc :latest (info-for-new-version))))))
 
-(defn- upload-version-info! []
-  (u/step "Upload version info"
-    (u/s3-copy! (format "s3://%s" (version-info-url)) (format "s3://%s.previous" (version-info-url)))
-    (u/s3-copy! (u/assert-file-exists (tmp-version-info-filename)) (format "s3://%s" (version-info-url)))
-    (u/create-cloudfront-invalidation! c/static-cloudfront-distribution-id (format "/%s" (version-info-filename)))))
+(defn- save-version-info!
+  [edition version-info]
+  (let [tmpname  (tmp-version-info-filename edition)]
+    (u/step (format "Delete and create %s" tmpname)
+      (u/delete-file-if-exists! tmpname)
+      (spit tmpname (json/generate-string version-info)))))
 
-(defn- validate-version-info []
-  (u/step (format "Validate version info at %s" (version-info-url))
-    (let [info           (current-version-info)
+(defn- upload-version-info!
+  ([edition]
+   (upload-version-info! edition false))
+  ([edition credentials-already-set?]
```

Comment:
I hate boolean flags like this but couldn't come up with a better idea. It somehow seemed grosser to parameterize this function with the copy function (i.e., `(defn upload-version-info! [edition copy-fn] ...)`)